collect_app/src/main/java/org/odk/collect/android/utilities/TimerLogger.java

```diff
@@ -0,0 +1,330 @@
+
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
+import android.os.SystemClock;
+
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.form.api.FormEntryController;
+import org.odk.collect.android.logic.FormController;
+import org.odk.collect.android.preferences.AdminKeys;
+import org.odk.collect.android.tasks.TimerSaveTask;
+
+import java.io.File;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+
+import timber.log.Timber;
+
+/**
+ * Handle logging of timer events and pass them to an Async task to append to a file
+ * Notes:
+ * 1) If the user has saved the form, then resumes editing, then exits without saving then the timing data during the
+ * second editing session will be saved.  This is OK as it records user activity.  However if the user exits
+ * without saving and they have never saved the form then the timing data is lost as the form editing will be
+ * restarted from scratch.
+ * 2) The times for questions in a group are not shown.  Only the time for the group is shown.
+ */
+public class TimerLogger {
+
+    public enum EventTypes {
+        FEC,                // FEC, Real type defined in FormEntryController
+        FORM_START,         // Start filling in the form
+        FORM_EXIT,          // Exit the form
+        FORM_RESUME,        // Resume filling in the form after previously exiting
+        FORM_SAVE,          // Save the form
+        FORM_FINALIZE,      // Finalize the form
+        HIERARCHY,          // Jump to a question
+        SAVE_ERROR,         // Error in save
+        FINALIZE_ERROR,     // Error in finalize
+        CONSTRAINT_ERROR    // Contraint or missing answer error on save
+    }
+
+    public class Event {
+
+        long start;
+        EventTypes eventType;
+        int fecType;
+        String node;
+        String dirn;
+
+        long end;
+        boolean endTimeSet;
+
+        private class EventDetails {
+            boolean hasIntervalTime;
+            String name;
+        }
+
+        /*
+         * Create a new event
+         */
+        Event(long start, EventTypes eventType, int fecType, String node, boolean advancingPage) {
+            this.start = start;
+            this.eventType = eventType;
+            this.fecType = fecType;
+            this.node = node;
+
+            if (eventType == EventTypes.FEC
+                    && (fecType == FormEntryController.EVENT_QUESTION
+                    || fecType == FormEntryController.EVENT_GROUP)) {
+                this.dirn = advancingPage ? "fwd" : "back";
+            } else {
+                this.dirn = "";
+            }
+
+            end = 0;
+            endTimeSet = false;
+        }
+
+        /*
+         * Return true if this is a view type event
+         *  Hierarchy Jump
+         *  Question
+         *  Prompt for repeat
+         */
+        public boolean isIntervalViewEvent() {
+            if (eventType == EventTypes.FEC
```

Comment:
You can just replace it with:

> return eventType == EventTypes.FEC
                    && (fecType == FormEntryController.EVENT_QUESTION
                    || fecType == FormEntryController.EVENT_GROUP
                    || fecType == FormEntryController.EVENT_END_OF_FORM
                    || fecType == FormEntryController.EVENT_PROMPT_NEW_REPEAT);